src/main/java/com/hp/application/automation/tools/run/SvExportBuilder.java

```diff
@@ -0,0 +1,143 @@
+
+    @DataBoundConstructor
+    public SvExportBuilder(String serverName, boolean force, String targetDirectory, boolean cleanTargetDirectory,
+                           SvServiceSelectionModel serviceSelection, boolean switchToStandByFirst) {
+        super(new SvExportModel(serverName, force, targetDirectory, cleanTargetDirectory, serviceSelection, switchToStandByFirst));
+    }
+
+    @Override
+    protected void logConfig(PrintStream logger, String prefix) {
+        logger.println(prefix + "Target Directory: " + model.getTargetDirectory());
+        logger.println(prefix + "Switch to Stand-By: " + model.isSwitchToStandByFirst());
+        super.logConfig(logger, prefix);
+    }
+
+    @Override
+    public DescriptorImpl getDescriptor() {
+        return (DescriptorImpl) super.getDescriptor();
+    }
+
+    @Override
+    public boolean performImpl(AbstractBuild<?, ?> build, Launcher launcher, BuildListener listener) throws Exception {
+        PrintStream logger = listener.getLogger();
+
+        ExportProcessor exportProcessor = new ExportProcessor(null);
+        IChmodeProcessor chmodeProcessor = new ChmodeProcessor(null);
+
+        ICommandExecutor exec = createCommandExecutor();
+
+        verifyNotNull(model.getTargetDirectory(), "Target directory must be set");
+
+        String targetDirectory = build.getWorkspace().child(model.getTargetDirectory()).getRemote();
+
+        if (model.isCleanTargetDirectory()) {
+            cleanTargetDirectory(logger, targetDirectory);
+        }
+
+        for (ServiceInfo serviceInfo : getServiceList(false, logger, build)) {
+            if (model.isSwitchToStandByFirst()) {
+                switchToStandBy(serviceInfo, chmodeProcessor, exec, logger);
+            }
+
+            logger.printf("  Exporting service '%s' [%s] to %s %n", serviceInfo.getName(), serviceInfo.getId(), targetDirectory);
+            verifyNotLearningBeforeExport(logger, exec, serviceInfo);
+            exportProcessor.process(exec, targetDirectory, serviceInfo.getId(), false);
+        }
+
+        return true;
+    }
+
+    private void verifyNotLearningBeforeExport(PrintStream logger, ICommandExecutor exec, ServiceInfo serviceInfo)
+            throws CommunicatorException, CommandExecutorException {
+
+        IService service = exec.findService(serviceInfo.getId(), null);
+        ServiceRuntimeConfiguration info = exec.getServiceRuntimeInfo(service);
+        if (info.getRuntimeMode() == ServiceRuntimeConfiguration.RuntimeMode.LEARNING) {
+            logger.printf("    WARNING: Service '%s' [%s] is in Learning mode. Exported model need not be complete!",
+                    serviceInfo.getName(), serviceInfo.getId());
+        }
+    }
+
+    private void switchToStandBy(ServiceInfo service, IChmodeProcessor chmodeProcessor, ICommandExecutor exec, PrintStream logger)
+            throws CommandExecutorException, SVCParseException, CommunicatorException {
```

Comment:
same as my previous comment - AbstractSVCException. You can ignore it or fix it by throwing just AbstractSVCException, even better to catch everyone with "AbstractSVCException" and log it (only 1 logger instead of 15 with one virtual method)